python/ql/consistency-queries/DataFlowConsistency.ql

```diff
@@ -0,0 +1,57 @@
+/**
+ * Provides consistency queries for checking invariants in the language-specific
+ * data-flow classes and predicates.
+ */
+
+private import python
+import semmle.python.dataflow.new.DataFlow::DataFlow
+private import semmle.python.dataflow.new.internal.DataFlowImplSpecific
+private import semmle.python.dataflow.new.internal.TaintTrackingImplSpecific
+private import codeql.dataflow.internal.DataFlowImplConsistency
+
+private module Input implements InputSig<PythonDataFlow> {
+  private import Private
+  private import Public
+
+  predicate argHasPostUpdateExclude(ArgumentNode n) {
+    exists(ArgumentPosition apos | n.argumentOf(_, apos) and apos.isStarArgs(_))
+    or
+    exists(ArgumentPosition apos | n.argumentOf(_, apos) and apos.isDictSplat())
```

Comment:
We have a similar exclusion in Ruby for implicit hash-splats, but this is only because we haven't yet implemented proper support. Is it the same for Python? I.e., in Ruby we do not currently handle
```rb
def foo(**args)
  args[:p].setField taint
end

foo(p: x)
sink(x.getField)
```

Reply:
Yes, the assertion in the following code holds. This was not really something I've thought too much about though, since it's not something I've ever seen in real code -- but I agree that it _could_ happen :+1:

```py
def set_foo(**args):
    args["p"].foo = 42

class MyClass: pass
c = MyClass()
set_foo(p=c)
assert c.foo == 42
```

Reply:
I have documented the missing flow for Ruby here: https://github.com/github/codeql/pull/14859.

Reply:
Nice, done for Python here: https://github.com/github/codeql/pull/14936